components/p3a/star_randomness_points.h

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_P3A_STAR_RANDOMNESS_POINTS_H_
+#define BRAVE_COMPONENTS_P3A_STAR_RANDOMNESS_POINTS_H_
+
+#include <memory>
+#include <string>
+
+#include "base/callback.h"
+#include "base/memory/ref_counted.h"
+#include "brave/components/p3a/constellation/rs/cxx/src/lib.rs.h"
+#include "brave/components/p3a/p3a_config.h"
+#include "brave/components/p3a/star_randomness_meta.h"
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+namespace p3a {
+
+struct P3AConfig;
+
+// Handles sending requests/handling responses to/from the randomness
+// server in order to receive randomness point data for STAR measurements.
+class StarRandomnessPoints {
+ public:
+  using RandomnessDataCallback = base::RepeatingCallback<void(
+      std::string histogram_name,
+      uint8_t epoch,
+      ::rust::Box<constellation::RandomnessRequestStateWrapper>
+          randomness_request_state,
+      std::unique_ptr<rust::Vec<constellation::VecU8>> resp_points,
+      std::unique_ptr<rust::Vec<constellation::VecU8>> resp_proofs)>;
+
+  StarRandomnessPoints(
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+      RandomnessDataCallback data_callback,
+      P3AConfig* config);
+  ~StarRandomnessPoints();
+
+  void SendRandomnessRequest(
+      std::string histogram_name,
+      StarRandomnessMeta* randomness_meta,
+      uint8_t epoch,
+      ::rust::Box<constellation::RandomnessRequestStateWrapper>
+          randomness_request_state,
+      const rust::Vec<constellation::VecU8>& rand_req_points);
+
+ private:
+  void HandleRandomnessResponse(
+      std::string histogram_name,
+      StarRandomnessMeta* randomness_meta,
+      uint8_t epoch,
+      ::rust::Box<constellation::RandomnessRequestStateWrapper>
+          randomness_request_state,
+      std::unique_ptr<std::string> response_body);
+
+  scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory_;
+  std::unique_ptr<network::SimpleURLLoader> url_loader_;
+
+  RandomnessDataCallback data_callback_;
+
+  P3AConfig* config_;
```

Comment:
don't use pointers to const data like this, prefer values or `const raw_ref<const P3AConfig>`